src/reason_parser.mly

```diff
@@ -2705,6 +2707,13 @@ _simple_expr:
         let loc = mklocation $symbolstartpos $endpos in
         bigarray_get ~loc $1 $4
       }
+  | simple_expr VARIABLEINFIXOP simple_expr
```

Comment:
You are making a rule of
```
simple_expr:
  | simple_expr VARIABLEINFIXOP simple_expr
```
`simple_expr` is not a right item here. Simple expression means a special kind of expression where you can just pass it as a function argument directly without wrapping them into parens. But that's not the case with infix operators.

We should put this under `expr`.